Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -0,0 +1,153 @@
+        let swizzledSelector = #selector(UIApplication.amp_sendAction)
+
+        guard
+            let originalMethod = class_getInstanceMethod(applicationClass, originalSelector),
+            let swizzledMethod = class_getInstanceMethod(applicationClass, swizzledSelector)
+        else { return }
+
+        let originalImp = method_getImplementation(originalMethod)
+        let swizzledImp = method_getImplementation(swizzledMethod)
+
+        class_replaceMethod(applicationClass,
+            swizzledSelector,
+            originalImp,
+            method_getTypeEncoding(originalMethod))
+        class_replaceMethod(applicationClass,
+            originalSelector,
+            swizzledImp,
+            method_getTypeEncoding(swizzledMethod))
+    }
+}
+
+extension UIApplication {
+    private var keyWindow: UIWindow? {
+        connectedScenes
+            .compactMap { $0 as? UIWindowScene }
+            .flatMap { $0.windows }
+            .first { $0.isKeyWindow }
+    }
+
+    @objc dynamic func amp_sendAction(_ action: Selector, to target: Any?, from sender: Any?, for event: UIEvent?) -> Bool {
+        let sendActionResult = amp_sendAction(action, to: target, from: sender, for: event)
+
+        guard
+            sendActionResult,
+            let keyWindow = keyWindow,
+            let view = sender as? UIView
+        else { return sendActionResult }
+
+        if let textField = view as? UITextField, !textField.shouldTrack(action, for: event) {
+            return sendActionResult
+        } else {
+            #if !os(tvOS)
+            if let slider = view as? UISlider, !slider.shouldTrack(action, for: event) {
+                return sendActionResult
+            }
+            #endif
+        }
+
+        let viewData = view.extractData(with: action, in: keyWindow)
+
+        UIKitUserInteractions.amplitudeInstances.allObjects.forEach {
+            $0.track(event: UserInteractionEvent(
+                viewController: viewData.viewController,
+                title: viewData.title,
+                accessibilityLabel: viewData.accessibilityLabel,
+                actionMethod: viewData.actionMethod,
+                targetViewClass: viewData.targetViewClass,
+                targetText: viewData.targetText,
+                hierarchy: viewData.hierarchy))
+        }
+
+        return sendActionResult
+    }
+}
+
+extension UIView {
+    private static let viewHierarchyDelimiter = " -> "
+
+    struct ViewData {
+        let viewController: String?
+        let title: String?
+        let accessibilityLabel: String?
+        let actionMethod: String
+        let targetViewClass: String
+        let targetText: String?
+        let hierarchy: String
+    }
+
+    func extractData(with action: Selector, in window: UIWindow) -> ViewData {
+        var targetText: String?
+
+        if let button = self as? UIButton {
+            targetText = button.titleLabel?.text
```

Comment:
So the UIAction passed to the control's initializer is not currently captured since UIControl doesn't pass that action to UIApplication.sendAction. One way to capture that as well is to swizzle UIControl.sendAction(:), which takes a UIAction and is triggered when a UIAction should be executed. I'm thinking that I can also change the current implementation to swizzle UIControl.sendAction(:to:for:) instead of UIApplication.sendAction. Any thoughts?

Reply:
IMO would save it for follow up unless the current approach is broken in extensions.